app/javascript/src/PaymentGateways/braintree/BraintreeForm.jsx

```diff
@@ -54,12 +56,23 @@ const BraintreeForm = ({
     }
   }, [braintreeNonceValue])
 
+  const get3DSecureError = (response) => {
+    return (
+      response.name === 'BraintreeError' ||
+        (
+          !response.threeDSecureInfo.status ||
+          !!response.threeDSecureInfo.status.match(/authenticate_(attempt_)?successful/) === false
```

Comment:
If we don't have `status` then we consier this an error? Is this the intended logic?

as a nit-pick with the logic above, the parentheses don't seem needed as well I didn't really understand why `!response.threeDSecureInfo.status.match(/authenticate_(attempt_)?successful/)` will not work.

Reply:
@akostadinov 

> If we don't have status then we consier this an error?

Yes, when OK the response returns `response.threeDSecureInfo.status` with either 'authenticate_successful' or 'authenticate_attempt_successful' value.

About the logic of !!, explanaition was lost with new commit, check [here](https://github.com/3scale/porta/pull/2574#discussion_r691014450)

Reply:
About the status, you can see this Braintree [doc](https://developer.paypal.com/braintree/docs/guides/3d-secure/server-side#status-codes):
Only "authenticate_attempt_successful" and "authenticate_successful" are accepted, there was also "authenticate_successful_issuer_not_participating" but is now deprecated.

Reply:
Can we have a test or two that will cover this logic?

Reply:
It's difficult to test this part, because those fields (credit card inputs) are injected by braintree in an iframe.
I wouldn't try to do it in this PR